Add `ConvectiveAdjustmentVerticalDiffusivity` in Docs (#2056)
CliMA/Oceananigans.jl: docs/src/physics/turbulence_closures.md

The constant ``C_b`` permits the "buoyancy modification" term it multiplies to be omitted
from a calculation.
By default we use the model constants ``C=1/12`` and ``C_b=0``.

## Convective adjustment vertical diffusivity

> This closure aims to model the enhanced mixing that occurs in the ocean's upper mixed layer due to convection.
> At every point and for every time instance, the closure checks the fluid's neutral stability and in any region
> that the buoyancy gradient is negative (unstable) it applies a vertical diffusivity `convective_κz` and or vertical
> viscosity `convective_νz` which are, usually, much enhanced compared to their normal values. That is,
> `convective_κz` ``\gg`` `background_κz` and `convective_νz` ``\gg`` `background_νz`.

Reviewer: ```suggestion
This closure aims to model the enhanced mixing that occurs in the ocean's upper mixed layer due to convection.
At every point and for every time instance, the closure diagnoses the gravitational stability of the fluid and applies the vertical diffusivities (i) `background_νz` to `u, v` and `background_κz` to all tracers if the fluid is gravitationally neutral or stable with `∂z(b) >= 0`, or (ii) `convective_νz` and `convective_κz` if `∂z(b) >= 0`.background_νz` and `background_κz`. This closure is a plausible model for convection if
`convective_κz` ``\gg`` `background_κz` and `convective_νz` ``\gg`` `background_νz`.
```

The constant ``C_b`` permits the "buoyancy modification" term it multiplies to be omitted
from a calculation.
By default we use the model constants ``C=1/12`` and ``C_b=0``.

## Convective adjustment vertical diffusivity

> This closure aims to model the enhanced mixing that occurs in the ocean's upper mixed layer due to convection.

Reviewer 2: ```suggestion
This closure aims to model the enhanced mixing that occurs due to convection.
```

Reviewer 2: I think this model is meant for  convective mixing in general, no?

> At every point and for every time instance, the closure diagnoses the gravitational stability of the fluid and applies the vertical diffusivities (i) `background_νz` to `u, v` and `background_κz` to all tracers if the fluid is gravitationally neutral or stable with `∂z(b) >= 0`, or (ii) `convective_νz` and `convective_κz` if `∂z(b) >= 0`.background_νz` and `background_κz`.

Reviewer 2: ```suggestion
At every point and for every time instance, the closure diagnoses the gravitational stability of the fluid and applies the vertical diffusivities (i) `background_νz` to `u, v` and `background_κz` to all tracers if the fluid is gravitationally neutral or stable with `∂z(b) >= 0`, or (ii) `convective_νz` and `convective_κz` if `∂z(b) >= 0`.
```
